src/main/java/uk/gov/hmcts/darts/usermanagement/service/impl/SecurityGroupServiceImpl.java

```diff
@@ -177,7 +179,38 @@ void updateSecurityGroupEntity(SecurityGroupPatch securityGroupPatch, SecurityGr
     }
 
     private void patchSecurityGroupUsers(SecurityGroupPatch securityGroupPatch, SecurityGroupEntity securityGroupEntity) {
+
         List<Integer> userIds = securityGroupPatch.getUserIds();
+        if (userIds != null) {
+            // get a list of system users for security group
+            List<Integer> systemUserIds = securityGroupEntity
+                .getUsers()
+                .stream()
+                .filter(user -> user.getIsSystemUser())
+                .map(UserAccountEntity::getId)
+                .toList();
+
+            // filter the incoming list to remove system users
+            List<UserAccountEntity> patchUsers = userAccountRepository
+                .findByIdIn(securityGroupPatch.getUserIds());
+
+            if (userIds.size() > 0 && patchUsers.isEmpty()) {
+                throw new DartsApiException(
+                    UserManagementError.USER_NOT_FOUND,
+                    String.format("No User accounts found for patch user IDs %s", securityGroupPatch.getUserIds()));
+            }
```

Comment:
Could you help me understand the purpose of this one?

Reply:
this is to check that the list of user ids passed in to the patch contains valid users. i added a comment in the code